linetools/spectra/xspectrum1d.py

```diff
@@ -702,39 +736,46 @@ def splice(self, spec2, wvmx=None, scale=1.):
         # Concatenate
         new_wv = np.concatenate((self.wavelength.value,
                                  spec2.wavelength.value[gdp]))
-        uwave = u.Quantity(new_wv, unit=self.wcs.unit)
+        uwave = u.Quantity(new_wv, unit=self.units['wave'])
         new_fx = np.concatenate((self.flux.value,
                                  spec2.flux.value[gdp] * scale))
-        if self.sig is not None:
+        if self.flag_sig:
             new_sig = np.concatenate((self.sig, spec2.sig[gdp] * scale))
+        else:
+            new_sig = None
         # Generate
-        spec3 = XSpectrum1D.from_array(
-            uwave, u.Quantity(new_fx), meta=self.meta.copy(),
-            uncertainty=StdDevUncertainty(new_sig))
+        spec3 = XSpectrum1D.from_tuple(
+            (uwave, u.Quantity(new_fx), new_sig), meta=self.meta.copy())
         # Return
         return spec3
 
     # Write to fits
     def write_to_ascii(self, outfil, format='ascii.ecsv'):
-        ''' Write to a text file.
+        """ Write to a text file.
 
         Parameters
         ----------
         outfil: str
           Filename.
-        '''
+        """
         # Convert to astropy Table
-        table = QTable([self.wavelength, self.flux, self.sig],
-                       names=('WAVE', 'FLUX', 'ERROR'))
+        table = QTable([self.wavelength, self.flux],
+                       names=('WAVE', 'FLUX'))
+        if self.flag_sig:
+            sigclm = Column(self.sig, name='ERROR')
+            table.add_column(sigclm)
+        if self.flag_co:
+            coclm = Column(self.co, name='CO')
+            table.add_column(coclm)
 
         # Write
         table.write(outfil, format=format)
 
     # Write to fits
-    def write_to_fits(self, outfil, clobber=True, add_wave=False):
-        ''' Write to a FITS file.
+    def write_to_fits(self, outfil, clobber=True):
+        """ Write to a FITS file.
 
-        Note that this does not generate a binary FITS table format.
+        This generates a binary FITS table format.
```

Comment:
Why make this a binary fits table? Can't we just stick with the old format but use 2d arrays for everything instead of 1d arrays?